pdp/static/js/prism_demo_map.js

```diff
@@ -69,37 +65,47 @@ function init_prism_map() {
     }
 
     function ncwms_params(layer_name) {
-        var varname = layer_name.split('/')[1];
-        var isClimatology = layer_name.split('/')[0].indexOf("Clim") !== -1;
-
-        if (varname === 'pr') {
-            this.params.LOGSCALE = true;
-            this.params.STYLES = 'boxfill/occam_inv';
-        } else {
-            this.params.LOGSCALE = false;
-            this.params.STYLES = 'boxfill/ferret';
+        // TODO: Place this utility function elsewhere. Note: Could use
+        //  lodash.get for this, but it would be more complicated.
+        function getWithDefault(obj, key, defKey="_default") {
+            return obj[key in obj ? key : defKey];
         }
 
-        if (varname === 'pr' && isClimatology) {
-            this.params.COLORSCALERANGE = '200,12500';
-        } else if (varname === 'pr') {
-            this.params.COLORSCALERANGE = '1,2000';
-        } else if (varname == 'tmax') {
-            this.params.COLORSCALERANGE = '-10,20';
-        } else if (varname == 'tmin' ) {
-            this.params.COLORSCALERANGE = '-15,10';
+        // Extract dataset unique id and variable name from layer name.
+        const layer_name_parts = layer_name.split('/');
+        const uniqueId = layer_name_parts[0];
+        const varname = layer_name_parts[1];
+
+        // Extract metadata from the unique id. This is ugly and fragile,
+        // but since PDP doesn't really have a data model, this is how this
+        // information is available to us here.
```

Comment:
yikes (but I agree we probably have to do it this way)

Reply:
All datasets that are published by the PDP are--by definition--also available from the `modelmeta` database. So we really don't have to extract metadata from the `unique_id` (in fact, I've been advocating against doing so for years). It may be locally *expedient*, but it's absolutely not *necessary*.

Reply:
Not, I think, without a pretty big change to this codebase.

Reply:
I definitely agree that it would require a broader change to the codebase. But both the comments in the code and in the PR say to me "the PDP can't do this any way but through the hacking the unique ID", and that's not true. So at the very least, I think that the code comment should point to the preferred solution that *is* available from the existing data model.

Reply:
It might be possible to use [the backend's existing DAS call](https://data.pacificclimate.org/data/gridded_observations/PNWNAmet_wind.nc.das) to get this sort of information in the future, since it reports any global metadata in the file. We'd definitely need to update a bunch of the PDP's datasets to do it, though; many of them predate the adoption of our current standards and have incomplete metadata.

Alternately, we might want to do something more like PCEX and add a new API endpoint that relays information from the modelmeta database.

Reply:
At the moment, I prefer the latter: https://github.com/pacificclimate/pdp/issues/193

Reply:
@jameshiebert , your point is well taken, particularly re. the comments. "Have to" is of course false; I intended it relative to the amount of effort it seemed reasonable to put into this particular PR. I've updated the code comments.